frontend/src/__tests__/basic.test.js

```diff
@@ -0,0 +1,246 @@
+    end_date: new Date("2019/12/31").toISOString(),
+    name: "First Session",
+    rate1: 56.54
+};
+
+const newPositionData = {
+    position_code: "MAT135F",
+    position_title: "Calculus I",
+    hours_per_assignment: 70,
+    start_date: new Date("2019/09/09").toISOString(),
+    end_date: new Date("2019/12/31").toISOString()
+};
+
+const newApplicantData = {
+    utorid: "weasleyr",
+    student_number: "89013443",
+    first_name: "Ron",
+    last_name: "Weasley",
+    email: "ron@potter.com",
+    phone: "543-223-9993"
+};
+
+/**
+ * Tests for the API. These are encapsulated in a function so that
+ * different `apiGET` and `apiPOST` functions can be passed in. For example,
+ * they may be functions that make actual requests via http or they may
+ * be from the mock API.
+ *
+ * @param {object} api
+ * @param {Function} api.apiGET A function that when passed a route will return the get response
+ * @param {Function} api.apiPOST A function that when passed a route and data, will return the post response
+ */
+function minimalSetupTest(api = { apiGET, apiPOST }) {
+    const { apiGET, apiPOST } = api;
+    let session = null;
+
+    beforeAll(async () => {
+        await apiPOST("/admin/debug/clear_data");
+    }, 15000);
+    afterAll(async () => {
+        await apiPOST("/admin/debug/snapshot");
+    });
+
+    let createdSessionId;
+    let createdPositionId;
+    let createdApplicantId;
+
+    it("create a session", async () => {
+        // get all the sessions so that we can check that our newly inserted session has
+        // a unique id.
+        const { payload: initialSessions } = await apiGET("/admin/sessions");
+
+        // check that there is no sessions before the create is run
+        expect(initialSessions).toMatchObject([]);
+
+        const resp1 = await apiPOST("/admin/sessions", newSessionData);
+        // check if we get a success response
+        expect(resp1).toMatchObject({ status: "success" });
+
+        const { payload: createdSession } = resp1;
+
+        // make sure the propTypes are right
+        checkPropTypes(sessionPropTypes, createdSession);
+        // make sure the data we get back is the same we put in
+        expect(createdSession).toMatchObject(newSessionData);
+        // make sure we have an id
+        expect(createdSession.id).not.toBeNull();
+
+        // fetch all sessions and make sure the data is in there
+        const { payload: withNewSessions } = await apiGET("/admin/sessions");
+        expect(withNewSessions.length).toBe(initialSessions.length + 1);
+        // make sure the id of our new session came back
+        expect(withNewSessions.map(x => x.id)).toContain(createdSession.id);
+
+        // save the id for later use
+        createdSessionId = createdSession.id;
+    });
+
+    it("create a position", async () => {
+        const { payload: initialPositions } = await apiGET(
+            `/sessions/${createdSessionId}/positions`
```

Comment:
Routes need to be prefixed with `/admin`. We might consider making a wrapper in `utils` that prefixes routes with a specific user so that it will be easier to test other users in the future.